core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -100,8 +101,8 @@ object ConsoleConsumer extends Logging {
         consumer.receive()
       } catch {
         case e: Throwable => {
-          error("Error processing message, stopping consumer: ", e)
-          consumer.stop()
+          error("Error processing message, terminating consumer process: ", e)
+          // Consumer will be stopped in shutdown hook
```

Comment:
The idea is that you want to dispose of resources as soon as you are done with them instead of waiting until an unrelated event happens (shutdown or GC for example). In some cases, the life of a resource may seem to be similar to the life of a particular tool (this is probably one such example). But that may no longer be the case if someone were to use `ConsoleConsumer` programmatically, for example.

The shutdown hook can be used as a back-up (like finalisation).